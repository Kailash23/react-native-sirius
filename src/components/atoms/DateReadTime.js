import React from 'react';
import { Text, View, Platform } from 'react-native';
import { localStrings } from '@utils/i18n';
import { withTheme } from 'react-native-paper';
import { SIZES, getFontSize } from '@styles/sizes';
import GetIcon from '@atoms/GetIcon';
import { customFonts } from '@styles/customFonts';

function DateReadTime(props) {
  const { formattedDate, readTime, style, theme, oldArticleWarning } = props;
  const color = oldArticleWarning
    ? theme.colors.warning
    : theme.colors.articleTimeStamp;
  return (
    <View>
      <View style={{ flexDirection: 'row' }}>
        {formattedDate && (
          <Text style={[style, { color: color }]}>
            {oldArticleWarning && (
              <GetIcon
                name={Platform.OS === 'ios' ? 'ios-warning' : 'md-warning'}
                size={getFontSize(SIZES.FONT)}
                color={color}
              />
            )}{' '}
            {formattedDate}
          </Text>
        )}
      </View>
      {readTime && (
        <Text
          style={{
            fontFamily: customFonts.merriweatherSansRegular,
            fontSize: getFontSize(SIZES.FONT),
            marginTop: 4,
            color: color
          }}>
          <GetIcon
            name={Platform.OS === 'ios' ? 'ios-timer' : 'md-time'}
            size={getFontSize(SIZES.FONT)}
            color={style.color}
          />
          {readTime
            ? ' ' + readTime + localStrings('authorByLine.readTime')
            : false}
        </Text>
      )}
    </View>
  );
}

export default withTheme(DateReadTime);
