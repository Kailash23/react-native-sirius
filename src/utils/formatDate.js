const moment = require('moment');

moment.updateLocale('en', {
  relativeTime: {
    h: '1 hour',
    d: '1 day'
  }
});

export const formatDateForAggregation = epochDate => {
  const date = new Date(epochDate);

  const now = moment();
  const moDate = moment.utc(date);

  const diff = now.diff(moDate, 'days');

  if (diff < 7) {
    return moDate.fromNow();
  } else {
    return moDate.format('MMMM D, YYYY');
  }
};

export const formatDateforArticle = epochDate => {
  const date = new Date(epochDate);

  const moDate = moment.utc(date);

  return moDate.format('dddd, MMMM D, YYYY');
};

export const isOverOneYearOld = epochDate => {
  const date = new Date(epochDate);
  const today = new Date();

  return date.getYear() > today.getYear();
};

export const formatDateforArticleAnalytics = epochDate => {
  const date = new Date(epochDate);

  const moDate = moment.utc(date);

  return moDate.format('ddd MMMM D YYYY');
};
