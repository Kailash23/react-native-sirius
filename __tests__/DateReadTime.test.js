import DateReadTime from '@atoms/DateReadTime';
import { formatDateforArticle, isOverOneYearOld } from '@utils/formatDate';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

configure({ adapter: new Adapter() });

describe('Testing <DateReadTime/>', () => {
  test('should render a DateReadTime with date and readtime ', () => {
    const wrapper = shallow(
      <DateReadTime
        formattedDate={formatDateforArticle(1592867400000)}
        readTime={'5'}
        oldArticleWarning={isOverOneYearOld(1592867400000)}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  describe('Format article date should work fine.', () => {
    test('formatDateforArticle(*) should return formatted date ', () => {
      expect(formatDateforArticle(1592867400000)).toBe('Monday, June 22, 2020');
    });
  });
  describe('Identifying Old article should work fine.', () => {
    test('isOverOneYearOld(*) should return a boolean', () => {
      expect(isOverOneYearOld(1592867400000)).toBe(false);
    });
  });
});
