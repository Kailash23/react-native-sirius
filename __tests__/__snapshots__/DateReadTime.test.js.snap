// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <DateReadTime/> should render a DateReadTime with date and readtime  1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  formattedDate="Monday, June 22, 2020"
  oldArticleWarning={false}
  readTime="5"
/>
`;
